packages/app/components/Queue/TA/TAAddStudent.tsx

```diff
@@ -48,15 +49,37 @@ export function AddStudentsModal({
     "Default grading type"
   );
   const [selectOptions, setSelectOptions] = useState([]);
+
+  const handleCheckboxChange = (id) => {
```

Comment:
I would probably leave comments explaining what this does and final structure of selectOptions look like. It's a bit hard to follow what's happening here